tests/crates/enarx_wasm_tests/src/bin/listen.rs

```diff
@@ -2,60 +2,116 @@
-    ensure!(
-        fd_count == 5, // STDIN, STDOUT, STDERR, the listening socket and a ping stream
-        "unexpected amount of file descriptors received"
-    );
-    ensure!(
-        env::var("FD_NAMES").context("failed to lookup `FD_NAMES`")?
-            == "stdin:stdout:stderr:ingest:ping"
-    );
+    fn assert_listener(port: NonZeroU16) -> anyhow::Result<()> {
+        eprintln!("[guest] listening on `{port}`");
+        let listener = File::options()
+            .read(true)
+            .write(true)
+            .open(format!("/net/lis/{port}"))
+            .map(OwnedFd::from)
+            .map(TcpListener::from)
+            .with_context(|| format!("failed to listen on port `{port}`"))?;
+
+        eprintln!("[guest] accepting on default listener");
+        let (stream, _) = listener
+            .accept()
+            .context("failed to accept connection on default listener stream")?;
+
+        eprintln!("[guest] asserting connectivity over default listener stream");
+        assert_stream(stream).context("failed to assert default listener stream")?;
+
+        listener
+            .set_nonblocking(false)
+            .context("failed to unset NONBLOCK")?;
+
+        eprintln!("[guest] accepting on blocking listener");
+        let (stream, _) = listener
+            .accept()
+            .context("failed to accept connection on blocking listener stream")?;
+
+        eprintln!("[guest] asserting connectivity over blocking listener stream");
+        assert_stream(stream).context("failed to assert blocking listener stream")?;
 
-    let listener = unsafe { TcpListener::from_raw_fd(3) };
-    let (stream, _) = listener
-        .accept()
-        .context("failed to accept first connection")?;
-    assert_stream(stream).context("failed to assert default stream")?;
-
-    listener
-        .set_nonblocking(false)
-        .context("failed to unset NONBLOCK")?;
-    let (stream, _) = listener
-        .accept()
-        .context("failed to accept second connection")?;
-    assert_stream(stream).context("failed to assert blocking stream")?;
-
-    listener
-        .set_nonblocking(true)
-        .context("failed to set NONBLOCK")?;
-    loop {
-        match listener.accept() {
-            Ok((stream, _)) => {
-                assert_stream(stream).context("failed to assert non-blocking stream")?;
-                break;
+        listener
+            .set_nonblocking(true)
+            .context("failed to set NONBLOCK")?;
+        loop {
+            eprintln!("[guest] accepting on non-blocking listener");
+            match listener.accept() {
+                Ok((stream, _)) => {
+                    eprintln!("[guest] asserting connectivity over non-blocking listener stream");
+                    assert_stream(stream)
+                        .context("failed to assert non-blocking listener stream")?;
+                    break;
+                }
+                Err(e) if e.kind() == io::ErrorKind::WouldBlock => {}
```

Comment:
This is a very busy loop now, running at 100% CPU

Reply:
I'll add a sleep